cmds/tcz/tcz.go

```diff
@@ -237,37 +196,24 @@ func setupPackages(tczName string, deps map[string]bool) error {
 		}
 
 		if err := os.MkdirAll(packagePath, dirMode); err != nil {
-			l.Fatalf("Package directory %s at %s, can not be created: %v", tczName, packagePath, err)
+			return fmt.Errorf("package directory %s at %s, can not be created: %v", tczName, packagePath, err)
 		}
 
-		loopname, err := findloop()
+		loopname, err := loop.FindDevice()
 		if err != nil {
-			l.Fatal(err)
+			return err
 		}
-		debug("findloop gets %v err %v\n", loopname, err)
 		pkgpath := filepath.Join(tczLocalPackageDir, v)
-		ffd, err := syscall.Open(pkgpath, syscall.O_RDONLY, 0)
-		if err != nil {
-			l.Fatalf("%v: %v\n", pkgpath, err)
-		}
-		lfd, err := syscall.Open(loopname, syscall.O_RDONLY, 0)
-		if err != nil {
-			l.Fatalf("%v: %v\n", loopname, err)
-		}
-		debug("ffd %v lfd %v\n", ffd, lfd)
-
-		a, b, errno := syscall.Syscall(SYS_ioctl, uintptr(lfd), LOOP_SET_FD, uintptr(ffd))
-		if errno != 0 {
-			l.Fatalf("loop set fd ioctl: pkgpath :%v:, loop :%v:, %v, %v, %v\n", pkgpath, loopname, a, b, errno)
+		if err := loop.SetFile(loopname, pkgpath); err != nil {
+			return err
 		}
 
 		/* now mount it. The convention is the mount is in /tinyCoreRoot/packagename */
-		if err := syscall.Mount(loopname, packagePath, "squashfs", syscall.MS_MGC_VAL|syscall.MS_RDONLY, ""); err != nil {
-			l.Fatalf("Mount :%s: on :%s: %v\n", loopname, packagePath, err)
+		if err := syscall.Mount(loopname, packagePath, "squashfs", syscall.MS_RDONLY, ""); err != nil {
```

Comment:
This seems to be the only user of "syscall" left. Given that the recommendation is to use `/x/sys/unix` instead, change this to unix.Mount I'd say.